### PR TITLE
improve_float_comparebranch

### PR DESCRIPTION
**Pull request by evemerak**
issue: #5 
Improve float_compare and float_branch functions.
These functions use too many instructions to check NaN, there is a easy way to check.

### CODE REVIEW

**Review by yadongw — APPROVED** [review]
lgtm

**Comment by zhengxiaolinX** [comment]
+1 - my vote may not count as required approving reviews, so just one comment.

File: src/hotspot/cpu/riscv64/macroAssembler_riscv64.cpp [hunk 1]
```diff
@@ -816,49 +816,45 @@ void MacroAssembler::la(Register Rd, Label &label) {
...
-      FLOATCMP##_d(t0, Rs1, Rs2);                                                                                \
-    }                                                                                                            \
-    bnez(t0, l, is_far);                                                                                         \
-  }
-
-  INSN(ble, fle);
-  INSN(blt, flt);
+#define INSN(NAME, FLOATCMP1, FLOATCMP2)                                              \
+  void MacroAssembler::float_##NAME(FloatRegister Rs1, FloatRegister Rs2, Label &l,   \
+                                    bool is_far, bool is_unordered) {                 \
```
Comment (kuaiwei) [inline src/hotspot/cpu/riscv64/macroAssembler_riscv64.cpp:821]: could you add comments for expected result just like MacroAssembler::FLOATTYPE##_compare

Comment (evemerak) [reply src/hotspot/cpu/riscv64/macroAssembler_riscv64.cpp:821]: I've added comments.

**Review by kuaiwei — APPROVED** [review]

**Pull request merged**